fe/fe-core/src/main/java/org/apache/doris/external/iceberg/IcebergCatalogImpl.java

```diff
@@ -14,54 +14,48 @@
 // KIND, either express or implied.  See the License for the
 // specific language governing permissions and limitations
 // under the License.
-
 package org.apache.doris.external.iceberg;
 
+import com.google.common.collect.Maps;
 import org.apache.doris.catalog.IcebergProperty;
-
 import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogUtil;
 import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Catalog;
 import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.SupportsNamespaces;
 import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-/**
- * HiveCatalog of Iceberg
- */
-public class HiveCatalog implements IcebergCatalog {
-    private static final Logger LOG = LogManager.getLogger(HiveCatalog.class);
+public class IcebergCatalogImpl implements IcebergCatalog {
+    private static final Logger LOG = LogManager.getLogger(IcebergCatalogImpl.class);
 
-    private org.apache.iceberg.hive.HiveCatalog hiveCatalog;
-
-    public HiveCatalog() {
-        hiveCatalog = new org.apache.iceberg.hive.HiveCatalog();
-    }
+    private String catalogType;
+    private Catalog icebergCatalog;
 
     @Override
     public void initialize(IcebergProperty icebergProperty) {
-        // set hadoop conf
+        this.catalogType = icebergProperty.getCatalogType();
+        Map<String, String> properties = Maps.newHashMap(icebergProperty.getExtraProperties());
+        properties.put(CatalogUtil.ICEBERG_CATALOG_TYPE, icebergProperty.getCatalogType());
+        // Hadoop configuration
         Configuration conf = new Configuration();
-        hiveCatalog.setConf(conf);
-        // initialize hive catalog
-        Map<String, String> catalogProperties = new HashMap<>();
-        catalogProperties.put("uri", icebergProperty.getHiveMetastoreUris());
-        hiveCatalog.initialize("hive", catalogProperties);
+        this.icebergCatalog = CatalogUtil.buildIcebergCatalog(icebergProperty.getCatalogType(), properties, conf);
```

Comment:
**Note**: `CatalogUtil.buildIcebergCatalog` only supports `HiveCatalog` and `HadoopCatalog` for now.
How do you plan to support other catalog types?

Reply:
We could use the `iceberg.catalog.catalog-impl` property to load other catalogs with the full catalog name.